GT-95 Reduce xml schema file loading for validating data entry values

ineilson opened the pull request:
Fixes #254

Including commit with cosmetic changes to calm PHP Mess detector warnings.

gregcorbett reviewed (changes requested):
Can you rebase this to resolve the conflicts?

ineilson commented:
Rebased to current dev and removed commit containing code standards updates to Site.php as it didn't contribute to the functionality of this pull request.
